src/types/engine_time.rs

```diff
@@ -0,0 +1,184 @@
+use std::time::Duration;
+
+use crate::raw::root::RED4ext as red;
+
+#[derive(Default, Clone, Copy)]
+#[repr(transparent)]
+pub struct EngineTime(red::EngineTime);
+
+impl EngineTime {
+    pub fn is_valid(&self) -> bool {
+        self.0.unk00 != [0; 8]
+    }
+
+    pub fn as_secs_f64(&self) -> f64 {
+        f64::from_ne_bytes(self.0.unk00)
+    }
+
+    /// SAFETY: panics if the sum ends up being `f64::NAN`, `f64::INFINITY` or `f64::NEG_INFINITY`.
```

Comment:
panics shouldn't be in a safety doc
```suggestion
    /// # Panics
    ///
    /// Panics if the sum ends up being `f64::NAN`, `f64::INFINITY` or `f64::NEG_INFINITY`.
```